cmds/legacy_boot/legacy_boot.go

```diff
@@ -0,0 +1,383 @@
+
+func get_supported_filesystem() []string {
+	var return_value []string
+	file, err := os.Open("/proc/filesystems")
+	check(err)
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	if err := scanner.Err(); err != nil {
+		log.Fatal(err)
+	}
+	var string1, string2 string
+	for scanner.Scan() {
+		string1 = ""
+		string2 = ""
+		fmt.Sscanf(scanner.Text(), "%s %s", &string1, &string2)
+		if string2 == "" {
+			if string1 != "" {
+				return_value = append(return_value, string1)
+			}
+		}
+	}
+	return return_value
+
+}
+
+// try to mount a specific block device
+
+func mount_entry(path string, supported_filesystem []string) bool {
+	var return_value bool
+	syscall.Mkdir("/u-root", 0777)
+	var flags uintptr
+	// Was supposed to be unecessary for kernel 4.x.x
+	if verbose {
+		println("/dev/" + path)
+	}
+	for _, filesystem := range supported_filesystem {
+		flags = syscall.MS_MGC_VAL
+		// Need to load the filesystem kind supported
+		syscall.Mkdir("/u-root/"+path, 0777)
+		err := syscall.Mount("/dev/"+path, "/u-root/"+path, filesystem, flags, "")
+		if err == nil {
+			return true
+		}
+	}
+	return_value = false
+	return return_value
+}
+
+func umount_entry(path string) bool {
+	var return_value bool
+	var flags int
+	// Was supposed to be unecessary for kernel 4.x.x
+	flags = syscall.MNT_DETACH
+	err := syscall.Unmount(path, flags)
+	if err == nil {
+		return true
+	}
+	return_value = false
+	return return_value
+}
+
+// This function is looking for grub.cfg file
+// and return absolute path to it
+
+func check_boot_entry(mount_point string) string {
+	_, err := os.Stat(mount_point + "/boot")
+	if err == nil {
+		// The boot directory is there
+		_, err2 := os.Stat(mount_point + "/boot" + "/grub")
+		if err2 == nil {
+			_, err3 := os.Stat(mount_point + "/boot" + "/grub" + "/grub.cfg")
+			if err3 == nil {
+				// found
+				return (mount_point + "/boot" + "/grub")
+			}
+		}
+	}
+	return ""
+
+}
+
+// This function is parsing a grub.cfg file
+// input: absolute directory path to grub.cfg
+// output: Return a list of strings with the following format
+//	 line[3*x] - menuconfig
+//	 line[3*x+1] - linux kernel + boot options
+// 	 line[3*x+2] - initrd
+// and the default boot entry configured into grub.cfg
+
+func get_file_menu_content(path string) ([]string, int) {
+	var return_value []string
+	file, _ := os.Open(path + "/grub.cfg")
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	if err := scanner.Err(); err != nil {
+		log.Fatal(err)
+	}
+	var status int
+	var int_return int
+	int_return = 0
+	status = 0
+	// When status = 0 we are looking for a menu entry
+	// When status = 1 we are looking for a linux entry
+	// When status = 2 we are looking for a initrd entry
+	var trimmed_line string
+	for scanner.Scan() {
+		trimmed_line = strings.TrimSpace(scanner.Text())
+		trimmed_line = strings.Join(strings.Fields(trimmed_line), " ")
+		if strings.HasPrefix(trimmed_line, "#") == false {
```

Comment:
Simplify to `if !strings.HasPrefix(trimmed_line, "#") {`, same for other strings.HasPrefix(...) == true|false below.

Reply:
Done I believe